arb_os/retrybuffer.mini

```diff
@@ -239,14 +245,20 @@ public impure func retryBuffer_delete(rbuf: RetryBuffer, txId: uint, withRefund:
         //         and attempts to delete it again will be no-ops
 
         if (withRefund) {
-            // credit the callvalue back to the beneficiary
-            evmCallStack_setAccountInCurrentContext(
-                rtx.beneficiary,
-                account_addToEthBalance(
-                    evmCallStack_getAccountInCurrentContext(rtx.beneficiary),
-                    rtx.callvalue,
-                )
-            );
+            // credit the escrowed callvalue back to the beneficiary
+            if (evmCallStack_isEmpty()) {
+                if let Some(astore) = accountStore_payFromEscrow(
+                    getGlobalAccountStore(),
+                    rtx.callvalueEscrowKey,
+                    rtx.beneficiary,
+                    rtx.callvalue
+                ) {
+                    setGlobalAccountStore(astore);
+                }  // else something went badly wrong and we can't refund the callvalue
+            } else {
+                // try to refund from the escrow, ignoring errors (which should never happen)
+                let _ = evmCallStack_transferEthFromEscrow(rtx.callvalueEscrowKey, rtx.beneficiary, rtx.callvalue);
```

Comment:
Maybe good to put a debugprint for debugging in case we have an error

Reply:
Added a debugprint